pkg/kubelet/pod/pod_manager_test.go

```diff
@@ -17,151 +17,273 @@ limitations under the License.
+			expectPod:         normalPod,
+			expectUID:         types.UID("normal-pod-uid"),
+			expectPodToMirrorMap: map[kubetypes.ResolvedPodUID]kubetypes.MirrorPodUID{
+				kubetypes.ResolvedPodUID("static-pod-uid"): kubetypes.MirrorPodUID("mirror-pod-uid"),
+			},
+			expectMirrorToPodMap: map[kubetypes.MirrorPodUID]kubetypes.ResolvedPodUID{
+				kubetypes.MirrorPodUID("mirror-pod-uid"): kubetypes.ResolvedPodUID("static-pod-uid"),
 			},
 		},
-	}
-	staticPod := &v1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			UID:         types.UID("static-pod-uid"),
-			Name:        "mirror-static-pod-name",
-			Namespace:   metav1.NamespaceDefault,
-			Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "file"},
+		{
+			name:              "Get static pod",
+			podList:           []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantUID:           types.UID("static-pod-uid"),
+			wantPod:           staticPod,
+			isMirrorPod:       false,
+			expectPods:        []*v1.Pod{staticPod, normalPod},
+			expectPod:         staticPod,
+			expectedMirrorPod: mirrorPod,
+			expectUID:         types.UID("static-pod-uid"),
+			expectPodToMirrorMap: map[kubetypes.ResolvedPodUID]kubetypes.MirrorPodUID{
+				kubetypes.ResolvedPodUID("static-pod-uid"): kubetypes.MirrorPodUID("mirror-pod-uid"),
+			},
+			expectMirrorToPodMap: map[kubetypes.MirrorPodUID]kubetypes.ResolvedPodUID{
+				kubetypes.MirrorPodUID("mirror-pod-uid"): kubetypes.ResolvedPodUID("static-pod-uid"),
+			},
 		},
-	}
-
-	expectedPods := []*v1.Pod{
 		{
-			ObjectMeta: metav1.ObjectMeta{
-				UID:         types.UID("extra-pod-uid"),
-				Name:        "extra-pod-name",
-				Namespace:   metav1.NamespaceDefault,
-				Annotations: map[string]string{kubetypes.ConfigSourceAnnotationKey: "api"},
+			name:              "Get mirror pod",
+			podList:           []*v1.Pod{mirrorPod, staticPod, normalPod},
+			wantUID:           types.UID("static-pod-uid"),
+			wantPod:           mirrorPod,
+			isMirrorPod:       true,
+			expectPods:        []*v1.Pod{staticPod, normalPod},
+			expectPod:         staticPod,
+			expectedMirrorPod: mirrorPod,
+			expectUID:         types.UID("static-pod-uid"),
+			expectPodToMirrorMap: map[kubetypes.ResolvedPodUID]kubetypes.MirrorPodUID{
+				kubetypes.ResolvedPodUID("static-pod-uid"): kubetypes.MirrorPodUID("mirror-pod-uid"),
+			},
+			expectMirrorToPodMap: map[kubetypes.MirrorPodUID]kubetypes.ResolvedPodUID{
+				kubetypes.MirrorPodUID("mirror-pod-uid"): kubetypes.ResolvedPodUID("static-pod-uid"),
 			},
 		},
-		staticPod,
 	}
-	updates := append(expectedPods, mirrorPod)
-	podManager, _ := newTestManager()
-	podManager.SetPods(updates)
+	for _, test := range testCase {
+		t.Run(test.name, func(t *testing.T) {
+			podManager := NewBasicPodManager()
+			podManager.SetPods(test.podList)
+			actualPods := podManager.GetPods()
+			assert.Empty(t, cmp.Diff(actualPods, test.expectPods, sortOpt), "actualPods and expectPods differ")
```

Comment:
nit:
```suggestion
			assert.Empty(t, cmp.Diff(test.expectPods, actualPods, sortOpt), "actualPods and expectPods differ")
```

The same should be done for pretty much all `cmp.Diff` calls in this file.

Reply:
done